samples/nrf9160/cloud_client/src/device_status_send.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <net/nrf_cloud.h>
+
+#include <logging/log.h>
+#include "cJSON.h"
+
+LOG_MODULE_REGISTER(cloud_status_update, CONFIG_CLOUD_CLIENT_LOG_LEVEL);
+
+#define SERVICE_INFO_JSON_NAME "serviceInfo"
+#define SERVICE_INFO_FOTA_VER_CURRENT 2
+#define SERVICE_INFO_FOTA_STR_BOOTLOADER "BOOT"
+#define SERVICE_INFO_FOTA_STR_MODEM "MODEM"
+#define SERVICE_INFO_FOTA_STR_APP "APP"
+#define UI_JSON_NAME "ui"
+#define FOTAS_JSON_NAME "fota_v"
+#define FOTAS_JSON_NAME_SIZE (sizeof(FOTAS_JSON_NAME) + 5)
+#define CLOUD_CHANNEL_STR_DEVICE_INFO "DEVICE"
+
+static int add_array_obj(const char * const items[], const uint32_t item_cnt,
+			 const char * const item_name, cJSON * const obj)
+{
+	cJSON *obj_to_add = NULL;
+	cJSON *str = NULL;
+
+	if ((obj == NULL) || (item_name == NULL)) {
+		return -EINVAL;
+	}
+
+	obj_to_add = cJSON_CreateArray();
+	if (obj_to_add == NULL) {
+		return -ENOMEM;
+	}
+
+	for (uint32_t cnt = 0; cnt < item_cnt; ++cnt) {
+		if (items[cnt] != NULL) {
+			str = cJSON_CreateString(items[cnt]);
+			if (str == NULL) {
+				cJSON_Delete(obj_to_add);
+				return -ENOMEM;
+			}
+			cJSON_AddItemToArray(obj_to_add, str);
+		}
+	}
+
+	/* if no strings were added, use NULL object */
+	if (cJSON_GetArraySize(obj_to_add) == 0) {
+		obj_to_add->type = cJSON_NULL;
+	}
+
+	cJSON_AddItemToObject(obj, item_name, obj_to_add);
+
+	return 0;
+}
+
+static int service_info_json_object_encode(
```

Comment:
https://github.com/nrfconnect/sdk-nrf/blob/c936aa6dca266c54baa663448aa35ea91f7c5d33/include/net/nrf_cloud.h#L500

Reply:
Good point.  I had copied this from Asset Tracker v1, which still must have its own local implementation.

Reply:
done